cub/util_type.cuh

```diff
@@ -39,12 +39,13 @@
 #include <limits>
 #include <type_traits>
 
-#if (__CUDACC_VER_MAJOR__ >= 9 || CUDA_VERSION >= 9000) && !_NVHPC_CUDA
+#include <cuda.h>
+
+#if !_NVHPC_CUDA
```

Comment:
While we're cleaning things up, seems like it'd make sense to add a `CUB_DISABLE_HALF_SUPPORT` for consistency with `CUB_DISABLE_BF16_SUPPORT`.

Reply:
I believe the reason this macro exists in the first place is the following [issue](https://github.com/NVIDIA/cub/issues/478):
> the bf16 headers are not redistributable according to the CUDA EULA

I don't know why someone would disable half support, so will skip this suggestion for now.

Reply:
Ah I didn't know the origin of that macro. How fun 🙂